tests/synthdnn/synthdnn.py

```diff
@@ -0,0 +1,97 @@
+
+    # Interface with benchdnn
+    matmul_parser.add_argument("benchdnn", nargs='?', help="path to benchdnn executable")
+    matmul_parser.add_argument("--engine", default="cpu", help="engine used for benchdnn execution")
+    matmul_parser.add_argument("--collect", default="corr", help="benchdnn collection type, can be one of [corr, perf]")
+    matmul_parser.add_argument("-n", "--name", default="", help="sample name")
+
+    # Sampler Arguments
+    matmul_parser.add_argument("-l", "--layouts", default="all",
+                               help="stag:wtag:dtag, comma separated list of layouts or \"all\" for every supported layout")
+    matmul_parser.add_argument("-m", "--iter-mode", default="zip", help ="iteration mode, must be one of zip or product")
+    matmul_parser.add_argument("-r", "--region", default="(1,1,1,1):(8,8192,8192,8192):(1,1,1,1)",
+                               help="(m_min,k_min,n_min)-(m_max,k_max,n_max)/(m_align,k_align, n_align)")
+    matmul_parser.add_argument("-s", "--samples", default=1000, help="number of samples to collect")
+    matmul_parser.add_argument("-t", "--types", default="all",
+                               help="dt:dt:dt(optional fpmath-mode), comma separated list of type configurations or \"all\" for every supported type")
+def matmul_main(args):
+    batchFile = open(args.batch_file, "w+t") if args.batch_file != None else None
+    if args.benchdnn != None and batchFile == None:
+        batchFile = NamedTemporaryFile("w+t")
+
+    if batchFile:
+        log(f"generating batch file: {batchFile.name}")
+        region = matmul_sampler.Region(args.region)
+        types = matmul.Types(args.types)
+        layouts = matmul.Layouts(args.layouts, region.ndims - 1)
+        samples= matmul_sampler.Sampler(int(args.samples), args.iter_mode,
+                                        types, layouts, region)
+        for s in samples:
+            batchFile.write("--reset " + s.benchdnn_str() + "\n")
+        batchFile.flush()
+        log(f"generation complete")
+
+    if args.benchdnn:
+        if not os.path.exists(args.benchdnn):
+            error(f"cannot execute {args.benchdnn}, no such file exists")
+
+        if(args.collect == "corr"):
+            benchdnn_args = f"--engine={args.engine} --matmul --mode-modifier=P"
+        elif(args.collect == "perf"):
+            benchdnn_args = f"--engine={args.engine} --matmul --mode=F --cold-cache=all --perf-template=sample,{args.name},%prb%,%0Gflops%,%0Gbw% --memory-kind=usm_device --attr-scratchpad=user"
+            if args.name.find(',') != -1:
+                error(f"sample name {args.name} contains invalid character: ,")
+        else:
+            error(f"unknown collection method {args.collect}")
+
+        cmd = f"{args.benchdnn} {benchdnn_args} --batch={batchFile.name}"
+        log(f"executing: {cmd}")
+        ret = os.system(cmd)
```

Comment:
In general, I think `subprocess.call` is better than `os.system`, and would eventually be required for report generation (to capture the output).

Reply:
I intend to leave this as is until `subprocess.call`  becomes necessary as I am attempting to minimize the script dependencies. By using subprocess, we increase the number of dependencies as the `os` modules is already used elsewhere.